test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -120,45 +127,85 @@ public String getID() {
         final Map<String, Object> pravegaPersistentVolumeSpec = getPersistentVolumeClaimSpec("20Gi", "standard");
+        
+        final String bookkeeperImg = DOCKER_REGISTRY + PREFIX + "/" + BOOKKEEPER_IMAGE_NAME;
+        final String pravegaImg = DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME;
+        final Map<String, Object> bookkeeperImgSpec;
+        final Map<String, Object> pravegaImgSpec;
+
+        if (LATER_OPERATOR) {
+
+            bookkeeperImgSpec = ImmutableMap.of("repository", bookkeeperImg);
+            pravegaImgSpec = ImmutableMap.of("repository", pravegaImg);
+
+        } else {
+
+            bookkeeperImgSpec = getImageSpec(bookkeeperImg, PRAVEGA_BOOKKEEPER_VERSION);
+            pravegaImgSpec = getImageSpec(pravegaImg, PRAVEGA_VERSION);
+
+        }
+
+        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder().put("image", bookkeeperImgSpec)
+                .put("replicas", bookieCount)
+                .put("resources", getResources("2000m", "5Gi", "1000m", "3Gi"))
+                .put("storage", ImmutableMap.builder()
+                        .put("indexVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .put("ledgerVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .put("journalVolumeClaimTemplate", bkPersistentVolumeSpec)
+                        .build())
+                .put("autoRecovery", true)
+                .build();
 
         final Map<String, Object> pravegaSpec = ImmutableMap.<String, Object>builder().put("controllerReplicas", controllerCount)
-                                                                                      .put("segmentStoreReplicas", segmentStoreCount)
-                                                                                      .put("debugLogging", true)
-                                                                                      .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
-                                                                                      //.put("options", props.getProperties())
-                                                                                      .put("options", props)
-                                                                                      .put("image",
-                                                                                       getImageSpec(DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME, PRAVEGA_VERSION))
-                                                                                      .put("tier2", tier2Spec())
-                                                                                      .build();
+                .put("segmentStoreReplicas", segmentStoreCount)
+                .put("debugLogging", true)
+                .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
+                .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
```

Comment:
Maybe it is advisable to merge https://github.com/pravega/pravega/pull/3980 before merging this PR, so we only focus on the compatibility issue in this PR.

Reply:
Since #3980 is not merged so far and resource changes are tested with operator 0.3.0 and operator 0.4.0 with this PR, I would prefer we merge this and close that PR.

Reply:
Both the issues can be handled from this PR itself hence I closed that PR